persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/field/MongoListField.scala

```diff
@@ -71,6 +72,18 @@ class MongoListField[OwnerType <: BsonRecord[OwnerType], ListType: Manifest](rec
       case list@c::xs if mf.runtimeClass.isInstance(c) => setBox(Full(list.asInstanceOf[MyType]))
       case Some(list@c::xs) if mf.runtimeClass.isInstance(c) => setBox(Full(list.asInstanceOf[MyType]))
       case Full(list@c::xs) if mf.runtimeClass.isInstance(c) => setBox(Full(list.asInstanceOf[MyType]))
+      case jlist: java.util.List[_] => {
+        if(!jlist.isEmpty) {
+          val elem = jlist.get(0)
+          if(elem.isInstanceOf[org.bson.Document]) {
+            setFromDocumentList(jlist.asInstanceOf[java.util.List[org.bson.Document]])
+          } else {
+            setBox(Full(jlist.toList.asInstanceOf[MyType]))
```

Comment:
well - maybe it can be done better - I'm open for suggestions. when I tried in scala REPL:
```
scala> :paste
// Entering paste mode (ctrl-D to finish)

val l = new java.util.ArrayList[Int](2)
l.add(1)
l.add(2)
import scala.collection.JavaConverters._
l.asScala

// Exiting paste mode, now interpreting.

l: java.util.ArrayList[Int] = [1, 2]
import scala.collection.JavaConverters._
res6: scala.collection.mutable.Buffer[Int] = Buffer(1, 2)
```